### PR TITLE
ipaddr migration to utils

### PR DESCRIPTION
**Pull request by ashwini-mhatre**
##### SUMMARY

Add deprecation warning in netcommon for ipaddr filters plugins 

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by Qalthos — CHANGES_REQUESTED** [review]
If we've moved all the ipaddr filters now, this should allow us to delete the ipaddr filter from netcommon now as well, unless there is something special about filter plugins that prevents this.

File: meta/runtime.yml
Comment (Qalthos) [inline meta/runtime.yml:66]: Shouldn't these have redirects?
```suggestion
        warning_text: Use 'ansible.utils.cidr_merge' module instead.
      redirect: ansible.utils.cidr_merge
```

**Comment by ashwini-mhatre** [comment]
> 

should i remove ipaddr.py file?

**Comment by ashwini-mhatre** [comment]
@Qalthos Tested it for one plugin its working as expected.
```
TASK [debug] *********************************************************************************************************************************************************************
task path: /Users/amhatre/ansible-collections/playbooks/test_ipwrap.yaml:17
Loading collection ansible.netcommon from /Users/amhatre/ansible-collections/collections/ansible_collections/ansible/netcommon
[DEPRECATION WARNING]: Use 'ansible.utils.ipwrap' module instead. This feature will be removed from ansible.netcommon in a release after 2024-01-01. Deprecation warnings can be 
disabled by setting deprecation_warnings=False in ansible.cfg.
redirecting filter ansible.netcommon.ipwrap to ansible.utils.ipwrap
Loading collection ansible.utils from /Users/amhatre/ansible-collections/collections/ansible_collections/ansible/utils
redirecting filter ansible.netcommon.ipwrap to ansible.utils.ipwrap
redirecting filter ansible.netcommon.ipwrap to ansible.utils.ipwrap
redirecting filter ansible.netcommon.ipwrap to ansible.utils.ipwrap
[WARNING]: The value '' is not a valid IP address or network, passing this value to ipaddr filter might result in breaking change in future.
redirecting (type: connection) ansible.builtin.network_cli to ansible.netcommon.network_cli
Loading collection cisco.iosxr from /Users/amhatre/ansible-collections/collections/ansible_collections/cisco/iosxr
<127.0.0.1> attempting to start connection
<127.0.0.1> using connection plugin ansible.netcommon.network_cli
Found ansible-connection at path /Users/amhatre/ansible_venvs/py3.6.10/bin/ansible-connection
<127.0.0.1> found existing local domain socket, using it!
[WARNING]: Persistent connection logging is enabled for 127.0.0.1. This will log ALL interactions and WILL NOT redact sensitive configuration like passwords. USE WITH CAUTION!
<127.0.0.1> updating play_context for connection
<127.0.0.1> 
<127.0.0.1> local domain socket path is /Users/amhatre/.ansible/pc/cbb067cabd
ok: [127.0.0.1] => {
    "msg": [
        "192.24.2.1",
        "host.fqdn",
        "[::1]",
        "",
        "192.168.32.0/24",
        "[fe80::100]/10",
        "[2001:db8:32c:faad::]/64",
        "True"
    ]
}
META: ran handlers
META: ran handlers

PLAY RECAP ***********************************************************************************************************************************************************************
127.0.0.1                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

(py3.6.10) amhatre@ashwinis-MacBook-Pro playbooks % 
```

File: changelogs/fragments/deprecate_ipaddr_filters.yaml
Comment (Qalthos) [inline changelogs/fragments/deprecate_ipaddr_filters.yaml:3]:
```suggestion
  - Redirected ipaddr filters to ansible.utils (https://github.com/ansible-collections/ansible.netcommon/pull/359).
```

**Review by Qalthos — CHANGES_REQUESTED** [review]
Looks great. My only concern is whether we need to hold this off until the major release in a few months, as this needs to depend on a minimum ansible.utils version that is not released yet.

**Comment by Qalthos** [comment]
recheck

**Review by Qalthos — CHANGES_REQUESTED** [review]
Looks like we need to remove the ipaddr unit tests as well. I assume those have moved to utils already?

**Comment by ashwini-mhatre** [comment]
Recheck

**Comment by ashwini-mhatre** [comment]
recheck

**Review by Qalthos — APPROVED** [review]

**Comment by Qalthos** [comment]
recheck

**Comment by Qalthos** [comment]
recheck

**Comment by ashwini-mhatre** [comment]
recheck

**Comment by ashwini-mhatre** [comment]
Recheck

**Comment by dtantsur** [comment]
FYI folks: it seems like this change breaks the non-namespaced usage of `nthhost` (without the collection). I understand it's probably deprecated, but just in case you did not realize it.

**Comment by jantari** [comment]
Also wanted to chime in and say this is a **breaking change** (and affected us), because the old `ansible.netcommon.ipaddr("network/prefix")` would return a single string but the new `ansible.utils.ipaddr("network/prefix")` returns a list containing a string.

Easy to verify by running the following task with both netcommon 2.5.1 and netcommon 2.6.0:

```yaml
- name: DEBUG NEW IPADDR FILTER
  debug:
    msg:
      - 'SUBNET VAR           : {{ item.subnet }}'
      - 'OLD NETCOMMON FILTER : {{ item.subnet | ansible.netcommon.ipaddr("network/prefix") }}'
      - 'NEW UTILS FILTER     : {{ item.subnet | ansible.utils.ipaddr("network/prefix") }}'
  loop:
    - subnet: 1.1.1.1/32
    - subnet: 10.10.10.0/24
    - subnet: 192.168.0.0/23
```

Output looks like this:

```
TASK [DEBUG IPADDR FILTER] *******************************************************
ok: [localhost] => (item={'subnet': '1.1.1.1/32'}) =>
  msg:
  - 'SUBNET VAR           : 1.1.1.1/32'
  - 'OLD NETCOMMON FILTER : 1.1.1.1/32'
  - 'NEW UTILS FILTER     : [''1.1.1.1/32'']'
ok: [localhost] => (item={'subnet': '10.10.10.0/24'}) =>
  msg:
  - 'SUBNET VAR           : 10.10.10.0/24'
  - 'OLD NETCOMMON FILTER : 10.10.10.0/24'
  - 'NEW UTILS FILTER     : [''10.10.10.0/24'']'
ok: [localhost] => (item={'subnet': '192.168.0.0/23'}) =>
  msg:
  - 'SUBNET VAR           : 192.168.0.0/23'
  - 'OLD NETCOMMON FILTER : 192.168.0.0/23'
  - 'NEW UTILS FILTER     : [''192.168.0.0/23'']'
```

**Comment by bluikko** [comment]
Was it intentional to make `| ipaddr()` not work without namespace/FQCN? How was a change like this done in a minor version update and not even listed in changelog?

**Comment by trishnaguha** [comment]
@dtantsur @bluikko The non-namespace breaking was not intentional. It is a bug. We are working on the FQCN fix now https://github.com/ansible/ansible/issues/77192.

**Comment by ashwini-mhatre** [comment]
We have released utils 2.5.2 and netcommon 2.6.1 which fixed https://github.com/ansible-collections/ansible.netcommon/issues/375 and https://github.com/ansible-collections/ansible.utils/issues/148 issues.

For FQDN issue ansible-core team already merged https://github.com/ansible/ansible/pull/77210. This fix will be available in 28th march release.

which means,
For non fqdn ipaddr filters users can use old netcommon collection version 2.5.1 as workaround until core-team will release ansible code(28th march). For fqdn ipaddr filters users can use latest versions of any of netcommon(2.6.1) or utils(2.5.2) cc: @trishnaguha

**Comment by blakedot** [comment]
Do you folks honestly expect everyone to rewrite the thousands of existing playbooks that use the ipaddr() filter?
Do you folks ever test your code on production configurations?

**Pull request merged**
